kroxylicious-filters/kroxylicious-record-validation/src/test/java/io/kroxylicious/proxy/filter/schema/JsonSchemaBytebufValidatorTest.java

```diff
@@ -98,63 +98,96 @@ public static void shutdownMockRegistry() {
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(true, Result::valid);
     }
 
-    private static Result validate(Record record, BytebufValidator validator) {
-        try {
-            return validator.validate(record.value(), record, false).toCompletableFuture().get(5, TimeUnit.SECONDS);
-        }
-        catch (ExecutionException | InterruptedException | TimeoutException e) {
-            throw new RuntimeException(e);
-        }
+    @Test
+    void valueWithWrongSchemaIdInHeaderRejected() {
+        Header[] headers = new Header[]{ new RecordHeader(SerdeHeaders.HEADER_VALUE_GLOBAL_ID, toByteArray(GLOBAL_ID + 1)) };
+        Record record = record(RECORD_KEY, VALID_JSON, headers);
+        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(apicurioConfig, GLOBAL_ID);
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(false, Result::valid)
+                .returns("Unexpected schema id in record (2), expecting 1", Result::errorMessage);
     }
 
-    private Record createRecord(String key, String value) {
-        ByteBuffer keyBuf = toBufNullable(key);
-        ByteBuffer valueBuf = toBufNullable(value);
-
-        try (ByteBufferOutputStream bufferOutputStream = new ByteBufferOutputStream(1000); DataOutputStream dataOutputStream = new DataOutputStream(bufferOutputStream)) {
-            DefaultRecord.writeTo(dataOutputStream, 0, 0, keyBuf, valueBuf, Record.EMPTY_HEADERS);
-            dataOutputStream.flush();
-            bufferOutputStream.flush();
-            ByteBuffer buffer = bufferOutputStream.buffer();
-            buffer.flip();
-            return DefaultRecord.readFrom(buffer, 0, 0, 0, 0L);
-        }
-        catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+    @Test
+    void valueWithWrongSchemaIdInBodyRejected() {
+        var value = asSchemaIdPrefixBuf(GLOBAL_ID + 1, VALID_JSON);
+        Record record = record(RECORD_KEY, value);
+        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(apicurioConfig, GLOBAL_ID);
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(false, Result::valid)
+                .returns("Unexpected schema id in record (2), expecting 1", Result::errorMessage);
     }
 
-    private static ByteBuffer toBufNullable(String key) {
-        if (key == null) {
-            return null;
-        }
-        byte[] keyBytes = key.getBytes(StandardCharsets.UTF_8);
-        return ByteBuffer.wrap(keyBytes);
+    private byte[] toByteArray(long globalId) {
+        var buf = ByteBuffer.allocate(Long.BYTES);
+        buf.putLong(globalId);
+        buf.flip();
```

Comment:
I was a bit mystified why this was flipped and `asSchemaIdPrefixBuf` isn't. I think the flip isn't needed as we are just grabbing the backing array immediately and discarding the buffer object.

Reply:
It is superfluous.  An earlier version of the method was returning a ByteBuffer.  I forgot to remove the flip.